### PR TITLE
Adopt :autosummary: to multiple modules

### PR DESCRIPTION
**Pull request by mthrok**
Adopt `:autosummary:` to various modules

    * torchaudio.compliance.kaldi
    * torchaudio.sox_effects
    * torchaudio.utils

### CODE REVIEW

**Comment by carolineechen** [comment]
overall I think this direction is good. a couple suggestions based on the pytorch core docs
- can we increase the column width of the function name (pytorch uses 50%)? currently a lot of the function names are being split into 2-3 lines which is not very readable
- for the subsections, can we also include section headers to more clearly label/separate the sections? right now there's no clear distinction, just a small gap between sections

**Comment by hwangjeff** [comment]
looks good overall, and i agree with @carolineechen's recommendations. another nit is that the "torchaudio.transforms" just before the bulleted list of sections in the main body seems redundant

File: docs/source/utils.rst [hunk 1]
```diff
@@ -1,21 +1,20 @@
...
 
-.. py:module:: torchaudio.utils
-
-torchaudio.utils.sox_utils
-~~~~~~~~~~~~~~~~~~~~~~~~~~
+``torchaudio.utils`` module contains utility functions to configure the global state of third party libraries.
 
-Utility module to configure libsox.
-This affects functionalities in :ref:`Sox IO backend<sox_io_backend>` and :ref:`Sox Effects<sox_effects>`.
+py:mod:`~torchaudio.utils.sox_utils` affects functionalities in :ref:`Sox IO backend<sox_io_backend>` and :ref:`Sox Effects<sox_effects>`.
```
Comment (carolineechen) [inline docs/source/utils.rst:8]: This is not output correctly https://output.circle-artifacts.com/output/job/bcd411e1-349b-48c0-a4ac-b8f704fec6ef/artifacts/0/docs/utils.html

File: docs/source/sox_effects.rst [hunk 2]
```diff
@@ -1,35 +1,31 @@
...
-
-.. autofunction:: init_sox_effects
-
-.. autofunction:: shutdown_sox_effects
-
-Listing supported effects
-~~~~~~~~~~~~~~~~~~~~~~~~~
-
-.. autofunction:: effect_names
+.. minigallery:: torchaudio.sox_effects.apply_effects_tensor torchaudio.sox_effects.apply_effects_file
```
Comment (carolineechen) [inline docs/source/sox_effects.rst:8]: this should be moved to the end of the section to stay consistent with the other modules formats. also there is some redundancy in the tutorials with those two functions which results in odd display: https://output.circle-artifacts.com/output/job/bcd411e1-349b-48c0-a4ac-b8f704fec6ef/artifacts/0/docs/sox_effects.html

```suggestion
.. minigallery:: torchaudio.sox_effects.apply_effects_file
```

**Review by carolineechen — APPROVED** [review]

**Review by nateanl — APPROVED** [review]

**Comment by carolineechen** [comment]
@mthrok is there also plans to adopt autosummary for the [torchaudio](https://pytorch.org/audio/main/torchaudio.html), [torchaudio.kaldi_io](https://pytorch.org/audio/main/kaldi_io.html), and [torchaudio.backend](https://pytorch.org/audio/main/backend.html) sections?

**Comment by mthrok** [comment]
> @mthrok is there also plans to adopt autosummary for the [torchaudio](https://pytorch.org/audio/main/torchaudio.html), [torchaudio.kaldi_io](https://pytorch.org/audio/main/kaldi_io.html), and [torchaudio.backend](https://pytorch.org/audio/main/backend.html) sections?

@carolineechen Not at the moment. It is a bit complicated for `torchaudio.backend` but should be easy for the rest. It's not in my immediate list of things.

**Pull request closed without merging**